src/app/dynamic_server/AccessControl.cpp

```diff
@@ -39,6 +44,11 @@ class DeviceTypeResolver : public Access::AccessControl::DeviceTypeResolver
 public:
     bool IsDeviceTypeOnEndpoint(DeviceTypeId deviceType, EndpointId endpoint) override
     {
+        // TODO: DynamicDispatcher.cpp actually hardcodes `false` here for every device/ep
+        //       we could remove the ember-compatibility-functions.h dependency if we hardcode
+        //       false here as well.
+        //
+        //       Should determine if hardcoding false here is the correct thing to do.
```

Comment:
It's not in general, once you try to support a non-hardcoded policy of any sort.  Right now all the hardcoding is restricted to DynamicDispatcher.cpp..

Reply:
I removed the comment. However I believe long-term the fact that dynamic_server includes ember-compatibility-functions.h but provides its own implementation is somewhat odd. I believe we have to make the compatibility functions as some sort of interface that people can override (since dynamic-dispatch does just that, except it does so at link-time).